packages/flutter/lib/src/rendering/wrap.dart

```diff
@@ -392,11 +391,10 @@ class RenderWrap extends RenderBox with ContainerRenderObjectMixin<RenderBox, Wr
     while (child != null) {
       final double childWidth = child.getMaxIntrinsicWidth(double.infinity);
       final double childHeight = child.getMaxIntrinsicHeight(childWidth);
-      if (runWidth + childWidth > width) {
-        height += runHeight;
-        if (runCount > 0)
-          height += runSpacing;
-        runCount += 1;
+      final double spaceWidth = childCount > 0 ? spacing : 0;
+      // child count check for situation when width of first child in run over available width
```

Comment:
This comment is a bit confusing, maybe `There must be at least one child before we move on to the next run.`.